src/model/geom/track.py

```diff
@@ -0,0 +1,52 @@
+from dataclasses import dataclass
+from typing import List, Tuple, Iterable
+
+from planar import Vec2
+from planar.line import LineSegment
+
+import utils
+from model.geom.segment import TrackSegment
+from model.geom.sensor import Sensor
+from model.geom.wall import Wall
+from utils import pairwise
+
+
+SegmentId = int
+
+
+@dataclass
+class Track:
+    segments: List[TrackSegment]
+
+    @classmethod
+    def from_points(cls, points: Iterable[Tuple[Vec2, Vec2]]) -> "Track":
+        segments: List[TrackSegment] = []
+        for (s1, s2), (e1, e2) in pairwise(points):
+            segments.append(
+                TrackSegment(
+                    Wall(LineSegment.from_points((s1, e1))),
+                    Wall(LineSegment.from_points((s2, e2))),
+                )
+            )
+
+        return cls(segments)
```

Comment:
Why Track class has auto-generated constructor and single @classmethod which functionality that could be moved to __init__? Is it any convention, some sort of simplification or simply we expect that we will add more @classmethods in the nearest future?

Reply:
The default method for construction a `Track` is a list of `TrackSegment`s, any other constructor can be written in terms of that one. Construction from a list of points is secondary to that most basic one.